atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -390,7 +404,66 @@ public RowColumnRangeIterator getRowsColumnRange(TableReference tableRef,
                         LoggingArgs.durationMillis(stopwatch)));
     }
 
-    private  <T> T maybeLog(Supplier<T> action, BiConsumer<LoggingFunction, Stopwatch> logger) {
+    private RowColumnRangeIterator wrapRowColumnRangeIteratorWithLogger(
+            Supplier<RowColumnRangeIterator> rowColumnRangeIteratorSupplier,
+            BiConsumer<LoggingFunction, Stopwatch> logger) {
+        RowColumnRangeIterator rowColumnRangeIterator = rowColumnRangeIteratorSupplier.get();
+        return new RowColumnRangeIterator() {
+            @Override
+            public Entry<Cell, Value> next() {
+                return maybeLog(rowColumnRangeIterator::next, logger);
```

Comment:
we probably want to update the log statement to indicate that this is one batch of the query - and maybe also indicate how many cells were in the batch

Reply:
The logger should have this information. Check above, where we construct it. I've made it more clear,